### DISCUSSION TITLE
End portals

### DISCUSSION THREAD

**Discussion by IceWaffles** [PurpurMC/Purpur#706] [__Dismissed__] [answered]
This is two suggestions regarding a improvement to end portals. Firstly, I believe it would be a nice-to-have if you could choose to not respawn at your bed when going through portals-a simple feature that would be massively useful towards me. Secondly, obviously servers have spawns where blocks cannot be placed. Players have been dropping boats through the portals in order to just cram the spawn with boats-so I suggest adding a feature to block anything other than players that can drop through end portals.

**Comment by BillyGalbreath** [comment, marked as answer]
> choose to not respawn at your bed when going through portals

this is a bit harder than it sounds. entering the end portal is not a normal teleport. from the server's perspective the player gets removed from the world and the client is sent a "win game" packet. the client then shows the end credits for 3 hours before talking back to the server saying it's spawning back in. since the player has no previous location their bed is used (or world spawn if no bed set). to change this behavior would change the behavior for all respawning, even just normal deaths. the server doesnt know the difference.

> block anything other than players that can drop through end portals

https://purpur.pl3x.net/docs/Configuration/#entities-can-use-portals

**Comment by IceWaffles** [comment]
I see; thanks for taking your time to reply. I still hope the portal respawn can be looked into, although I know understand it'll be difficult and time consuming if it will be done at all.
